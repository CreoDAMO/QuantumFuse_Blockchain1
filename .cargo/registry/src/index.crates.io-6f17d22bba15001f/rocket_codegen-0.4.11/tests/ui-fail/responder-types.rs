// normalize-stderr-test: "<(.*) as (.*)>" -> "$1 as $$TRAIT"
// normalize-stderr-test: "and \d+ others" -> "and $$N others"
// normalize-stderr-test: "::: .*\.rs" -> "::: $$FILE.rs"

#![feature(proc_macro_hygiene, decl_macro)]

#[macro_use] extern crate rocket;

#[derive(Responder)]
struct Thing1 {
    thing: u8,
    //~^ ERROR Responder
}

#[derive(Responder)]
struct Thing2 {
    thing: String,
    other: u8,
    //~^ ERROR Header
}

#[derive(Responder)]
struct Thing3 {
    thing: u8,
    //~^ ERROR Responder
    other: u8,
    //~^ ERROR Header
}

#[derive(Responder)]
struct Thing4 {
    thing: String,
    other: ::rocket::http::ContentType,
    then: String,
    //~^ ERROR Header
}

#[get("/")]
fn foo() -> usize { 0 }
//~^ ERROR Responder

fn main() {  }
