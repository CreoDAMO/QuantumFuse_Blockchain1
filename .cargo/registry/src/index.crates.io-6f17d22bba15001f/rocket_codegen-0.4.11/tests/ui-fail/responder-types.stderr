error[E0277]: the trait bound `u8: Responder<'_>` is not satisfied
  --> tests/ui-fail/responder-types.rs:11:5
   |
11 |     thing: u8,
   |     ^^^^^^^^^ the trait `Responder<'_>` is not implemented for `u8`
   |
   = help: the following other types implement trait `Responder<'r>`:
             <&'r [u8] as Responder<'r>>
             <&'r str as Responder<'r>>
             <() as Responder<'r>>
             <Content<R> as Responder<'r>>
             <File as Responder<'r>>
             <Flash<R> as Responder<'r>>
             <JavaScript<R> as Responder<'r>>
             <MsgPack<R> as Responder<'r>>
           and 30 others

error[E0277]: the trait bound `rocket::http::Header<'_>: std::convert::From<u8>` is not satisfied
   --> tests/ui-fail/responder-types.rs:18:5
    |
18  |     other: u8,
    |     ^^^^^^^^^ the trait `std::convert::From<u8>` is not implemented for `rocket::http::Header<'_>`
    |
    = help: the following other types implement trait `std::convert::From<T>`:
              <rocket::http::Header<'static> as std::convert::From<&'a Cookie<'c>>>
              <rocket::http::Header<'static> as std::convert::From<AcceptCharset>>
              <rocket::http::Header<'static> as std::convert::From<AcceptEncoding>>
              <rocket::http::Header<'static> as std::convert::From<AcceptLanguage>>
              <rocket::http::Header<'static> as std::convert::From<AcceptRanges>>
              <rocket::http::Header<'static> as std::convert::From<AccessControlAllowCredentials>>
              <rocket::http::Header<'static> as std::convert::From<AccessControlAllowHeaders>>
              <rocket::http::Header<'static> as std::convert::From<AccessControlAllowMethods>>
            and 39 others
    = note: required because of the requirements on the impl of `Into<rocket::http::Header<'_>>` for `u8`
note: required by a bound in `rocket::Response::<'r>::set_header`
   --> $WORKSPACE/core/lib/src/response/response.rs
    |
    |     pub fn set_header<'h: 'r, H: Into<Header<'h>>>(&mut self, header: H) -> bool {
    |                                  ^^^^^^^^^^^^^^^^ required by this bound in `rocket::Response::<'r>::set_header`

error[E0277]: the trait bound `u8: Responder<'_>` is not satisfied
  --> tests/ui-fail/responder-types.rs:24:5
   |
24 |     thing: u8,
   |     ^^^^^^^^^ the trait `Responder<'_>` is not implemented for `u8`
   |
   = help: the following other types implement trait `Responder<'r>`:
             <&'r [u8] as Responder<'r>>
             <&'r str as Responder<'r>>
             <() as Responder<'r>>
             <Content<R> as Responder<'r>>
             <File as Responder<'r>>
             <Flash<R> as Responder<'r>>
             <JavaScript<R> as Responder<'r>>
             <MsgPack<R> as Responder<'r>>
           and 30 others

error[E0277]: the trait bound `rocket::http::Header<'_>: std::convert::From<u8>` is not satisfied
   --> tests/ui-fail/responder-types.rs:26:5
    |
26  |     other: u8,
    |     ^^^^^^^^^ the trait `std::convert::From<u8>` is not implemented for `rocket::http::Header<'_>`
    |
    = help: the following other types implement trait `std::convert::From<T>`:
              <rocket::http::Header<'static> as std::convert::From<&'a Cookie<'c>>>
              <rocket::http::Header<'static> as std::convert::From<AcceptCharset>>
              <rocket::http::Header<'static> as std::convert::From<AcceptEncoding>>
              <rocket::http::Header<'static> as std::convert::From<AcceptLanguage>>
              <rocket::http::Header<'static> as std::convert::From<AcceptRanges>>
              <rocket::http::Header<'static> as std::convert::From<AccessControlAllowCredentials>>
              <rocket::http::Header<'static> as std::convert::From<AccessControlAllowHeaders>>
              <rocket::http::Header<'static> as std::convert::From<AccessControlAllowMethods>>
            and 39 others
    = note: required because of the requirements on the impl of `Into<rocket::http::Header<'_>>` for `u8`
note: required by a bound in `rocket::Response::<'r>::set_header`
   --> $WORKSPACE/core/lib/src/response/response.rs
    |
    |     pub fn set_header<'h: 'r, H: Into<Header<'h>>>(&mut self, header: H) -> bool {
    |                                  ^^^^^^^^^^^^^^^^ required by this bound in `rocket::Response::<'r>::set_header`

error[E0277]: the trait bound `rocket::http::Header<'_>: std::convert::From<std::string::String>` is not satisfied
   --> tests/ui-fail/responder-types.rs:34:5
    |
34  |     then: String,
    |     ^^^^^^^^^^^^ the trait `std::convert::From<std::string::String>` is not implemented for `rocket::http::Header<'_>`
    |
    = help: the following other types implement trait `std::convert::From<T>`:
              <rocket::http::Header<'static> as std::convert::From<&'a Cookie<'c>>>
              <rocket::http::Header<'static> as std::convert::From<AcceptCharset>>
              <rocket::http::Header<'static> as std::convert::From<AcceptEncoding>>
              <rocket::http::Header<'static> as std::convert::From<AcceptLanguage>>
              <rocket::http::Header<'static> as std::convert::From<AcceptRanges>>
              <rocket::http::Header<'static> as std::convert::From<AccessControlAllowCredentials>>
              <rocket::http::Header<'static> as std::convert::From<AccessControlAllowHeaders>>
              <rocket::http::Header<'static> as std::convert::From<AccessControlAllowMethods>>
            and 39 others
    = note: required because of the requirements on the impl of `Into<rocket::http::Header<'_>>` for `std::string::String`
note: required by a bound in `rocket::Response::<'r>::set_header`
   --> $WORKSPACE/core/lib/src/response/response.rs
    |
    |     pub fn set_header<'h: 'r, H: Into<Header<'h>>>(&mut self, header: H) -> bool {
    |                                  ^^^^^^^^^^^^^^^^ required by this bound in `rocket::Response::<'r>::set_header`

error[E0277]: the trait bound `usize: Responder<'_>` is not satisfied
   --> tests/ui-fail/responder-types.rs:39:13
    |
39  | fn foo() -> usize { 0 }
    |             ^^^^^ the trait `Responder<'_>` is not implemented for `usize`
    |
    = help: the following other types implement trait `Responder<'r>`:
              <&'r [u8] as Responder<'r>>
              <&'r str as Responder<'r>>
              <() as Responder<'r>>
              <Content<R> as Responder<'r>>
              <File as Responder<'r>>
              <Flash<R> as Responder<'r>>
              <JavaScript<R> as Responder<'r>>
              <MsgPack<R> as Responder<'r>>
            and 30 others
note: required by a bound in `handler::<impl Outcome<rocket::Response<'r>, rocket::http::Status, rocket::Data>>::from`
   --> $WORKSPACE/core/lib/src/handler.rs
    |
    |     pub fn from<T: Responder<'r>>(req: &Request, responder: T) -> Outcome<'r> {
    |                    ^^^^^^^^^^^^^ required by this bound in `handler::<impl Outcome<rocket::Response<'r>, rocket::http::Status, rocket::Data>>::from`
